### ISSUE TITLE
ELK logging or metrics

### ISSUE THREAD

**Issue by praveenkumar** [projectatomic/adb-utils#86]
Currently we don't have any ELK (ElasticSearch Logging) or metrics (like hawkular) enabled which will be good for developer experience.

**Comment by LalatenduMohanty** [comment]
@praveenkumar @hferentschik sorry I am still not able to get context of the use case. Not sure how it is useful from CDK/ADB point of view.

**Comment by hferentschik** [comment]
It is useful for the OpenShift user. The whole idea is to give a developer the chance to experience a as broad as possible functionality range of OpenShift out of the box. Configuring persistence volumes falls into the same category. When giving presentations, one of the questions which comes up is: "How do you do monitoring? Are there centralized logs? ..."

Personally, I am provisioning my CDK instance with Hawkular for demos. Partly to show the metrics in the console and partly it is a requirement for pod auto scaling which I like to demo.

So if possible I would love to see unified logging and metrics part as out of the box experience. However, I am a bit concerned about startup times (more containers need to come up, starting Hawkular takes quite some time on my machine) and resource usage. Especially the latter is important. Running the all-in-one OpenShift on a developer machine is quite a stretch to begin with. By adding all these other things we are even putting more load onto the VM. For that reason I think it would be great if things like metrics and logging are tested and optionally available. That would be the best imo. The provision scripts for these things are all there, but don't run per default. By setting and option you can then enable metrics resp. logging. I think any user who wants to experiment with these things would appreciate it, since working out the right provisioning sequence is sometimes quite tricky.

Last but not least, another reason to not enable Hawkular per default is that it works very flaky. At least on our all-in-on setup. I am not sure whether this is related to the all-in-one approach or maybe due to resource problems, but before I'd enable it out of the box, it would need more testing imo.

Bottom line, definitely a +1 for enabling (centralized) logging and metrics, but it should be optionally via flag or so (maybe this is the right use for the OPENSHIFT_OPTIONS variable we had in the startup script).

**Comment by alberttwong** [comment]
Just wanted to post that I was the one that asked for this originally on the container-tools mailing list.

**Comment by jorgemoralespou** [comment]
[This](https://github.com/openshift-evangelists/vagrant-origin/blob/11c3be8d8e9dda7f574f7322b7b4f7339528baa9/scripts/base/addons-setup#L125-L179) is how we install logging and it works

**Comment by praveenkumar** [comment]
Thanks @jorgemoralespou to point out, we are currently have some base code refactoring to add those capabilities as addons.
